sklearn/datasets/samples_generator.py

```diff
@@ -754,35 +760,73 @@ def make_blobs(n_samples=100, n_features=2, centers=3, cluster_std=1.0,
     """
     generator = check_random_state(random_state)
 
-    if isinstance(centers, numbers.Integral):
-        centers = generator.uniform(center_box[0], center_box[1],
-                                    size=(centers, n_features))
+    if isinstance(n_samples, int):
+        # n_centers is figured by the centers arg
+        if centers is None:
+            centers = 3
+
+        if isinstance(centers, numbers.Integral):
+            centers = generator.uniform(center_box[0], center_box[1],
```

Comment:
just set `n_centers` here (and above) and reuse the generation and checking code below